src/app/identity.service.ts

```diff
@@ -262,7 +262,25 @@ export class IdentityService {
 
     return firstValueFrom(this.http.get(uri).pipe(map(((response: any) => response.token))));
   }
+
+  /**
+   * After a non-logged in donor donates they have an account but there is no password set on it. This method
+   * allows setting a password using the secret number that we emailed to them in their donation confirmation
+   * message.
+   */
+  async setFirstPasswordWithToken(password: string, {person_uuid, secretNumber}: EmailVerificationToken): Promise<Object> {
+    const uri = `${environment.identityApiPrefix}/people/setFirstPassword`;
```

Comment:
`${environment.identityApiPrefix}/people/:uuid/setFirstPassword` is a bit more consistent with our approach elsewhere, but not a blocker. (Should have mentioned on ID PR sorry)